Lib/html/parser.py

```diff
@@ -243,7 +240,7 @@ def goahead(self, end):
                 assert 0, "interesting.search() lied"
         # end while
         if end and i < n and not self.cdata_elem:
-            if self.convert_charrefs and not self.cdata_elem:
+            if self.convert_charrefs:
```

Comment:
As above, can you elaborate on this part and check if there is a test for these paths?

Reply:
@ezio-melotti according to the diff, you added the (unnecessary) check in 2013: https://github.com/python/cpython/commit/95401c5f6b9f07b094924559177c9b30a1c38998#diff-da26f443eb16696a27ae56d506f3350528284855872c7fc98cb5a9231550ead8R305

Reply:
Looks like a straightforward cut an paste error in Ezio's changeset.

Reply:
Seems like the bug was in the line above, fixed here: https://github.com/python/cpython/commit/e76ff560b015a585e185d9a21d3c18540c6b5fcd#diff-da26f443eb16696a27ae56d506f3350528284855872c7fc98cb5a9231550ead8R260-L263

Thanks to @serhiy-storchaka who tracked down a 5-year-old PR to fix that bug and merged it